source/extensions/common/crypto/utility.cc

```diff
@@ -71,12 +84,21 @@ const VerificationOutput Utility::verifySignature(absl::string_view hash, EVP_PK
   return {false, absl::StrCat("Failed to verify digest. Error code: ", ok)};
 }
 
-PublicKeyPtr Utility::importPublicKey(const std::vector<uint8_t>& key) {
+CryptoObjectPtr importPublicKey(const std::vector<uint8_t>& key) {
   CBS cbs({key.data(), key.size()});
-  return PublicKeyPtr(EVP_parse_public_key(&cbs));
+
+  EVP_PKEY* pkey(EVP_parse_public_key(&cbs));
+
+  auto publicKeyWrapper = new PublicKeyObject();
+  publicKeyWrapper->setEVP_PKEY(pkey);
+
+  std::unique_ptr<PublicKeyObject> publicKeyPtr = std::make_unique<PublicKeyObject>();
+  publicKeyPtr.reset(publicKeyWrapper);
+
+  return publicKeyPtr;
```

Comment:
Can you not do `return std::make_unique<PublicKeyObject>(publicKeyWrapper)`?

Reply:
local identifier should be `snake_case` (throughout the PR)